kernel/nfft/nfft.c

```diff
@@ -2478,6 +2478,12 @@ static void nfft_adjoint_1d_B(X(plan) *ths)
 
 void X(trafo_1d)(X(plan) *ths)
 {
+  if((ths->N[0] <= ths->m) || (ths->n[0] <= 2*ths->m+2))
```

Comment:
As far as I see from the `nfft.c` code, e.g., inside the function `static void nfft_trafo_1d_compute` there is a `for` loop:
```
  if (u < o)
  {
    for (l = 1, gj = g + u, (*fj) = (*psij++) * (*gj++); l <= 2*m+1; l++)
      (*fj) += (*psij++) * (*gj++);
  }
```
The arrays `g` and `g_hat` in the `nfft_plan` have size small `n` for the one-dimensional case. For `n < 2 * m + 1`, there might be a buffer overflow (invalid read for nfft_trafo, invalid write for nfft_adjoint).